Automattic/jetpack
# My Jetpack: Add trial support through Search

Fixes #27018 

#### Changes proposed in this Pull Request:

* Add support for `has_trial_support` method, which changes status for `needs_purchase_or_free`.
* Add check for `pricing_for_ui.trial_available` flag in `ProductDetailCard` which points checkout url for `wpcomFreeProductSlug` and includes `Start for free` button.
* Receive `checkoutUrl` in `clickHandler` at `ProductInterstitial`.

#### Other information:


#### Jetpack product discussion


p1666615281220519-slack-C02ME06LF

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






* Start `My Jetpack`
* Insert `new_pricing_202208=1` search param in URL.
* You should see the new flow when search is installed or even when it isn't installed.
